src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -0,0 +1,389 @@
+        public override void SetHeader(string headerName, string headerValue)
+        {
+            if (contentHeaderNames.Contains(headerName))
+            {
+                // If this is a "Content" header then we cache the value (due
+                // to the message content not being set yet) and set it
+                // upon sending the message (when the content will be available)
+                this.contentHeaderValueCache[headerName] = headerValue;
+                return;
+            }
+
+            this.requestMessage.Headers.Remove(headerName);
+            this.requestMessage.Headers.Add(headerName, headerValue);
+        }
+
+        /// <summary>
+        /// Gets the stream to be used to write the request payload.
+        /// </summary>
+        /// <returns>Stream to which the request payload needs to be written.</returns>
+        public override Stream GetStream()
+        {
+            return this.messageStream;
+        }
+
+        /// <summary>
+        /// Abort the current request.
+        /// </summary>
+        public override void Abort()
+        {
+            this.client.CancelPendingRequests();
+        }
+
+        /// <summary>
+        /// Begins an asynchronous request for a System.IO.Stream object to use to write data.
+        /// </summary>
+        /// <param name="callback">The System.AsyncCallback delegate.</param>
+        /// <param name="state">The state object for this request.</param>
+        /// <returns>An System.IAsyncResult that references the asynchronous request.</returns>
+        public override IAsyncResult BeginGetRequestStream(AsyncCallback callback, object state)
+        {
+            var taskCompletionSource = new TaskCompletionSource<Stream>();
+            taskCompletionSource.TrySetResult(this.messageStream);
+            return taskCompletionSource.Task.ToApm(callback, state);
+        }
+
+        /// <summary>
+        /// Ends an asynchronous request for a System.IO.Stream object to use to write data.
+        /// </summary>
+        /// <param name="asyncResult">The pending request for a stream.</param>
+        /// <returns>A System.IO.Stream to use to write request data.</returns>
+        public override Stream EndGetRequestStream(IAsyncResult asyncResult)
+        {
+            return ((Task<Stream>)asyncResult).Result;
+        }
+
+        public override IAsyncResult BeginGetResponse(AsyncCallback callback, object state)
+        {
+            var send = CreateSendTask();
+            return send.ToApm(callback, state);
+        }
+
+        public override IODataResponseMessage EndGetResponse(IAsyncResult asyncResult)
+        {
+            return UnwrapAggregateException(() =>
+            {
+                var result = ((Task<HttpResponseMessage>)asyncResult).Result;
+                return ConvertHttpClientResponse(result);
+            });
+        }
+
+#if !PORTABLELIB && !(NETCOREAPP1_0 || NETCOREAPP2_0)

```

Comment:
Any reason we are still worrying about portablelib?

Reply:
resolved